src/main/scala/com/actian/spark_vector/colbuffer/ColumnBufferFactory.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 Actian Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.actian.spark_vector.colbuffer
+
+import org.apache.spark.Logging
+
+import com.actian.spark_vector.colbuffer.integer._
+import com.actian.spark_vector.colbuffer.real._
+import com.actian.spark_vector.colbuffer.decimal._
+import com.actian.spark_vector.colbuffer.singles._
+import com.actian.spark_vector.colbuffer.string._
+import com.actian.spark_vector.colbuffer.time._
+import com.actian.spark_vector.colbuffer.timestamp._
+
+/** This is a `Factory` implementation of `ColumnBuffers`. */
+object ColumnBufferFactory extends Logging {
```

Comment:
1. By doing matches, either within the same block in the (future) ColumnBuffer or by composing/reducing partial function such as 
`def build: PartialFunction[Boolean, ColumnBuffer[]] = { case condition => new instance }`, will force us to repeat the conditional expression, i.e. case side, for some types (e.g. Decimal, Time, Timestamp, etc) which likely would have only 1-2 params that differ (e.g. scaleMin-Max, precisionMin-Max). In other cases is the getNewInstance which is common among particular data types, see Timestamp{LZ,NZ,TZ}{Long,LongLong} types. So, actually, the inheritance of supportColumnType and/or getNewInstance with overwritten vals show more usefulness.

2. We talked 1:1 and we conclude that it doesn't work because the _.register_ needs to be explicitly called at runtime. Is not enough to do the ColumnBuffer.register(build) within the trait, right?